### PR TITLE
refactor: caching fines

### PR DESCRIPTION
**Pull request by MoskalykA**

### CODE REVIEW

**Comment by Gellipapa** [comment]
Hi! @MoskalykA Thank you for pr, but i don't understand what is this. Please describe what you have modified and how to test it.A little more than that "refactor".

**Comment by MoskalykA** [comment]
Hi, this allows you to make a single SQL query and reduce the number of nets sent. It's a kind of small optimization

File: client/main.lua [hunk 1]
```diff
@@ -521,35 +521,44 @@ function OpenFineMenu(player)
...
-			{unselectable = true, icon = "fas fa-scroll", title = TranslateCap('fine')}
+	if not fineList[category] then
+		ESX.TriggerServerCallback('esx_policejob:getFineList', function(fines)
+			fineList[category] = fines
+		end, category)
+
+		while fineList[category] == nil do
+			Wait(50)
+		end	
+	end
```
Comment (CsokiHUN) [inline client/main.lua:534]: if possible do not use a while loop

```suggestion
if not fineList[category] then
		local p = promise.new()

		ESX.TriggerServerCallback('esx_policejob:getFineList', function(fines)
			p:resolve(fines)
		end, category)

		fineList[category] = Citizen.Await(p)
end
```

**Review by CsokiHUN — CHANGES_REQUESTED** [review]

File: server/main.lua [hunk 2]
```diff
@@ -197,11 +197,18 @@ ESX.RegisterServerCallback('esx_policejob:getOtherPlayerData', function(source,
...
+		MySQL.query('SELECT * FROM fine_types WHERE category = ?', {category},
+		function(fines)
+			fineList[category] = fines
+
+			cb(fines)
+		end)
+	else
+		cb(fineList[category])
+	end
 end)
```
Comment (CsokiHUN) [inline server/main.lua:212]: you can load them all into fineList when the script starts

**Comment by Gellipapa** [comment]
@MoskalykA  Please change default sql files lang to EN. (Replace file)

**Comment by Gellipapa** [comment]
image

File: server/main.lua (hunk 2 above)
Comment (MoskalykA) [reply server/main.lua:212]: Yes, but that would be one request too many if nobody uses it x)

**Comment by MoskalykA** [comment]
Good 👍🏻

File: server/main.lua (hunk 2 above)
Comment (Gellipapa) [reply server/main.lua:212]: Hi! I think what @CsokiHUN  means is that then you don't need this callback, but you pre-load the data into a table and work with it and don't start a query to the DB every getFineList.

**Review by Gellipapa — APPROVED** [review]

**Pull request merged**